DotnetFoundation/DotnetFoundation.Api/Controllers/UserController.cs

```diff
@@ -84,17 +122,38 @@ public async Task<ActionResult<BaseResponse<bool>>> AddUserRoleAsync(UserRoleReq
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<BaseResponse<UserResponse>>> UpdateUserAsync(int userId, [FromBody] UpdateUserRequest updateUserRequest)
+    public async Task<ActionResult<BaseResponse<UserResponse>>> UpdateUserAsync(int userId, UpdateUserRequest updateUserRequest)
     {
         BaseResponse<UserResponse> response = new(ResponseStatus.Fail);
         if (updateUserRequest == null)
         {
             return BadRequest("Invalid request data");
         }
-        response.Data = await _userService.UpdateUserAsync(userId, updateUserRequest).ConfigureAwait(false);
-        response.Status = ResponseStatus.Success;
+        try
+        {
+            response.Data = await _userService.UpdateUserAsync(userId, updateUserRequest).ConfigureAwait(false);
+            response.Status = ResponseStatus.Success;
 
-        return Ok(response);
+            return Ok(response);
+        }
+        catch (NotFoundException ex)
+        {
+            response.Message = ex.Message;
+            response.Status = ResponseStatus.Error;
+            return BadRequest(response);
+        }
+        catch (UserException ex)
```

Comment:
I don't think this is required. If an exception is caused by the server/database operation, then it is not a BadRequest. It has to be Internal server error. Also, it can be handled in the catch block below.

Reply:
removed